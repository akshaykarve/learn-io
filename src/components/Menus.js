import React from 'react';
import './ComponentStyle.css';
import menusIcon from './images/menus.png';
const menus=({onRouteChange, route, menusClick, isSignedIn,isAdmin})=>{
		if(menusClick&&(route==='signin'||route==='register')){
			return(
				<div className='menusStyle'>
					<a href="#home" className='homeButton'>Learn-io</a>
					<div className='dropdown'>
						<button onClick={()=>onRouteChange('menus')} className='disableButton'><img src={menusIcon} height='30px' width='30px' alt="menus"/></button>
					</div>
				</div>
			);
		}else if(isAdmin){
			return(
				<div className='menusStyle'>
					<a href="#home" className='homeButton'>Learn-io</a>
					<div className='dropdown'>
						<button className='dropbtn'><img src={menusIcon} height='30px' width='30px' alt="menus"/></button>
						<div className='dropdown-content'>
							<a href="#2">Admin</a>
						    <a href="#2">My Profile</a>
						    <a href="#2">Setting</a>
						    <a href="#3">Your Page</a>
						    <a href="#3">Sign Out</a>
						</div>
					</div>
				</div>
			);
		}else if(isSignedIn){
			return(
				<div className='menusStyle'>
					<a href="#home" className='homeButton'>Learn-io</a>
					<div className='dropdown'>
						<button className='dropbtn'><img src={menusIcon} height='30px' width='30px' alt="menus"/></button>
						<div className='dropdown-content'>
						    <a href="#2">My Profile</a>
						    <a href="#2">Setting</a>
						    <a href="#3">Your Page</a>
						    <a href="#3">Sign Out</a>
						</div>
					</div>
				</div>
			);
		}else{
			return(
				<div className='menusStyle'>
					<a href="#home" className='homeButton'>Learn-io</a>
					<div className='dropdown'>
						<button className='dropbtn'><img src={menusIcon} height='30px' width='30px' alt="menus"/></button>
						<div className='dropdown-content'>
							<a href="#signin" onClick={()=>onRouteChange('signin')}>Sign In</a>
						</div>
					</div>
				</div>
			);
		}
		
}

export default menus;