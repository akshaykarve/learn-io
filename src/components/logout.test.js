import React from 'react';
import renderer from 'react-test-renderer';
import {default as LogoutController} from "./Menus";

const BrowserRouter = require("react-router-dom").BrowserRouter;
describe("Log Out Controller", ()=>{
    it('Log Out', () => {
        
        const tree = renderer.create(
        <BrowserRouter>
            <LogoutController
            />
        </BrowserRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
