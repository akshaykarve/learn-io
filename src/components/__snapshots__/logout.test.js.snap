// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Log Out Controller Log Out 1`] = `
<div
  className="menusStyle"
>
  <a
    className="homeButton"
    href="/home"
    onClick={[Function]}
  >
     Learn-io 
  </a>
  <div
    className="dropdown"
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      className="dropbtn dropdown-toggle btn btn-primary"
      disabled={false}
      id="dropdown-basic"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="menus"
        height="40px"
        src="menus.png"
        width="40px"
      />
    </button>
  </div>
</div>
`;
